contrib/gimp.lua

```diff
@@ -0,0 +1,154 @@
+      exported image
+    * Press "export"
+    * Edit the image with GIMP then save the changes with File->Overwrite....
+    * Exit GIMP
+    * The edited image will be imported and grouped with the original image
+
+    CAVEATS
+    * Developed and tested on Ubuntu 14.04 LTS with darktable 2.0.3 and GIMP 2.9.3 (development version with
+      > 8 bit color)
+    * There is no provision for dealing with the xcf files generated by GIMP, since darktable doesn't deal with 
+      them.  You may want to save the xcf file if you intend on doing further edits to the image or need to save 
+      the layers used.  Where you save them is up to you.
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("gimp",dt.configuration.config_dir.."/lua/")
+
+local function get_filename(str)
+  -- Thanks to Tobias Jakobs for the awesome regular expression
+  local path, filename, filetype = string.match(str, "(.-)([^\\/]-%.?([^%.\\/]*))$")
+  return filename
+end
+
+local function _(msgid)
+    return gettext.dgettext("gimp", msgid)
+end
+
+local function checkIfBinExists(bin)
+  local handle = io.popen("which "..bin)
+  local result = handle:read()
+  local ret
+  handle:close()
+  if (result) then
+    dt.print_error("true checkIfBinExists: "..bin)
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+local function show_status(storage, image, format, filename,
+  number, total, high_quality, extra_data)
+    dt.print(string.format(_("Export Image %i/%i"), number, total))
+end
+
+local function gimp_edit(storage, image_table, extra_data) --finalize
+  if not checkIfBinExists("gimp") then
+    dt.print_error(_("GIMP not found"))
+    return
+  end
+
+  -- list of exported images 
+  local img_list
+
+   -- reset and create image list
+  img_list = ""
+
+  for _,exp_img in pairs(image_table) do
+    img_list = img_list ..exp_img.. " "
+  end
+
+  dt.print(_("Launching GIMP..."))
+
+  local gimpStartCommand
+  gimpStartCommand = "gimp "..img_list
+  
+  dt.print_error(gimpStartCommand)
+
+  coroutine.yield("RUN_COMMAND", gimpStartCommand)
+
+  -- for each of the image, exported image pairs
+  --   move the exported image into the directory with the original  
+  --   then import the image into the database which will group it with the original
+  --   and then copy over any tags other than darktable tags
+
+  for image,exported_image in pairs(image_table) do
+
+    local myimage_name = image.path .. "/" .. get_filename(exported_image)
+
+    dt.print_error("moving " .. exported_image .. " to " .. myimage_name)
+    result = os.rename(exported_image, myimage_name)
```

Comment:
The way it stands now, if you exported a raw to gimp as a .jpg, then exported the resulting .jpg to gimp, the .jpg would get overwritten when gimp exits.  I could do the xxxx.jpg, then xxxx_01.jpg.  The only question is if I export _01.jpg does it become _02.jpg or _01_01.jpg?

I also developed a shortcut for the occasion when you did want to overwrite the file, i.e. I exported and edited, but forgot to clone out one more thing so I'll just reopen it and fix it.  While running that I discovered an unfortunate darktable "feature" (bug).  If I've not opened the image in darkroom, then dropping the cache causes it to regenerate the thumbnail and it shows correctly.  If I open the image in darkroom, it does not reflect the changes, so it seems that darkroom also maintains a cache.  When you return to lighttable, the thumbnail gets regenerated again and shows the image without any changes, so it appears to regenerate from the darkroom cache.  If you open it with gimp again and make more changes then save it, the thumbnail cache gets dropped, but it regenerates the image from the darkroom cache and not the file.  The only way to get things synced up is to restart darktable.